master/internal/api_experiment.go

```diff
@@ -43,3 +53,55 @@ func (a *apiServer) GetExperiments(
 	a.sort(resp.Experiments, req.OrderBy, req.SortBy, apiv1.GetExperimentsRequest_SORT_BY_ID)
 	return resp, a.paginate(&resp.Pagination, &resp.Experiments, req.Offset, req.Limit)
 }
+
+func (a *apiServer) PreviewSearch(
+	_ context.Context, req *apiv1.PreviewSearchRequest) (*apiv1.PreviewSearchResponse, error) {
+	bytes, err := protojson.Marshal(req.Config)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "error parsing experiment config: %s", err)
+	}
+	config := model.DefaultExperimentConfig()
+	if err = json.Unmarshal(bytes, &config); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "error parsing experiment config: %s", err)
+	}
+	if err = check.Validate(config.Searcher); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid experiment config: %s", err)
+	}
+
+	s := searcher.NewSearcher(
+		req.Seed, searcher.NewSearchMethod(config.Searcher), config.Hyperparameters)
+	sim, err := searcher.Simulate(s, nil, searcher.RandomValidation, true, config.Searcher.Metric)
+	if err != nil {
+		return nil, err
+	}
+	protoSim := &experimentv1.ExperimentSimulation{Seed: req.Seed}
+	indexes := make(map[string]int)
+	toProto := func(k searcher.Kind) experimentv1.WorkloadKind {
+		switch k {
+		case searcher.RunStep:
+			return experimentv1.WorkloadKind_WORKLOAD_KIND_RUN_STEP
+		case searcher.ComputeValidationMetrics:
+			return experimentv1.WorkloadKind_WORKLOAD_KIND_COMPUTE_VALIDATION_METRICS
+		case searcher.CheckpointModel:
+			return experimentv1.WorkloadKind_WORKLOAD_KIND_CHECKPOINT_MODEL
+		default:
+			return experimentv1.WorkloadKind_WORKLOAD_KIND_UNSPECIFIED
+		}
+	}
+	for _, result := range sim.Results {
+		var workloads []experimentv1.WorkloadKind
+		for _, msg := range result {
+			w := toProto(msg.Workload.Kind)
+			workloads = append(workloads, w)
+		}
+		hash := fmt.Sprint(workloads)
```

Comment:
honestly this got me a little lost:

1) it's not a hash technically
2) I'm not sure what we're trying to do here